content/webapp/components/OnThisPageAnchors/index.tsx

```diff
@@ -1,40 +1,156 @@
-import { FunctionComponent } from 'react';
+import { FunctionComponent, useState, useEffect } from 'react';
+import { useActiveAnchor } from '@weco/common/hooks/useActiveAnchor';
 import styled from 'styled-components';
 
-import { font } from '@weco/common/utils/classnames';
+import { font, FontFamily } from '@weco/common/utils/classnames';
 import PlainList from '@weco/common/views/components/styled/PlainList';
 import Space from '@weco/common/views/components/styled/Space';
 import { Link } from '@weco/content/types/link';
 
-const Anchor = styled.a.attrs({
+const ListItem = styled.li<{ $active?: boolean; $sticky?: boolean, activeColor?: string }>`
+  ${props => props.$sticky ? `
+  position: relative;
+  padding-left: 12px;
+  padding-bottom: 6px;
+  padding-top: 6px;
+  &::before {
+    content: '';
+    display: block;
+    position: absolute;
+    left: ${props.$active ? '0px' : '1px'};
+    top: 0;
+    bottom: 0;
+    width: ${props.$active ? '3px' : '1px'};
+    background: ${props.$active ? props.activeColor : props.theme.color('black')};
+  }
+` : ''}
+`;
+
+
+const Anchor = styled.a.attrs<{
+  $active?: boolean;
+  $backgroundBlend?: boolean;
+  $sticky?: boolean;
```

Comment:
np; For booleans we like the naming convention to be a question, so like `isActive`, `isSticky`, `hasBackgroundBlend` etc.